src/components/Popup/Popup.tsx

```diff
@@ -2,73 +2,78 @@ import * as React from 'react'
 import * as PropTypes from 'prop-types'
 import _ from 'lodash'
 import { Popper, PopperChildrenProps } from 'react-popper'
-import rtlCSSJS from 'rtl-css-js'
 
-import { childrenExist, customPropTypes, UIComponent, IRenderResultConfig } from '../../lib'
+import { childrenExist, AutoControlledComponent, IRenderResultConfig } from '../../lib'
 import { ItemShorthand, Extendable, ReactChildren } from '../../../types/utils'
-import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
-import Portal, { TriggerAccessibility } from '../Portal'
+
+import Ref from '../Ref'
+import computePopupPlacement, { Alignment, Position } from './positioningHelper'
+
 import PopupContent from './PopupContent'
+
 import { PopupBehavior } from '../../lib/accessibility'
 import {
   Accessibility,
   AccessibilityActionHandlers,
   IAccessibilityBehavior,
 } from '../../lib/accessibility/interfaces'
-import computePopupPlacement, { Alignment, Position, Placement } from './positioningHelper'
 
 const POSITIONS: Position[] = ['above', 'below', 'before', 'after']
 const ALIGNMENTS: Alignment[] = ['top', 'bottom', 'start', 'end', 'center']
 
 export interface IPopupProps {
   accessibility?: Accessibility
   align?: Alignment
-  as?: any
-  basic?: boolean
   children?: ReactChildren
   className?: string
   content?: ItemShorthand | ItemShorthand[]
+  defaultOpen?: boolean
+  open?: boolean
+  onOpenChange?: () => void
   position?: Position
   trigger?: JSX.Element
-  styles?: ComponentPartStyle
-  variables?: ComponentVariablesInput
 }
 
 export interface IPopupState {
   triggerRef: HTMLElement
-  popupOpened?: boolean
 }
 
 /**
  * A Popup displays additional information on top of a page.
  * @accessibility This is example usage of the accessibility tag.
  * This should be replaced with the actual description after the PR is merged
  */
-export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupState> {
+export default class Popup extends AutoControlledComponent<Extendable<IPopupProps>, IPopupState> {
   public static displayName = 'Popup'
 
   public static className = 'ui-popup'
 
   public static Content = PopupContent
 
   public static propTypes = {
+    /** Accessibility behavior if overridden by the user. */
+    accessibility: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
     /** Alignment for the popup. */
     align: PropTypes.oneOf(ALIGNMENTS),
 
-    /** An element type to render as (string or function). */
-    as: customPropTypes.as,
-
-    /** Basic CSS styling for the popup. */
-    basic: PropTypes.bool,
-
     /** The popup content (deprecated). */
-    children: customPropTypes.disallow(['children']),
+    children: PropTypes.any, // customPropTypes.disallow(['children']),
```

Comment:
is it intended?

Reply:
to remove `disallow` statement - yes, this is intended :) but at the same time (other than cleaning up commented code), PropTypes.node would be a better substitute here - let me address that.

Reply:
why do we remove it? we support children API now?

Reply:
yes - as this is quite reasonable for the Popup case, see:

```jsx
<SomeTree>
  <Popup content={popupContent}>
    <button>Click me to trigger popup</button>
  </Popup>
</SomeTree>
```

Now to introduce popup functionality it is necessary just to wrap the component that previously was in the tree, to augment it with popup functionality (and make it a trigger)